File: authorization-manager-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ScimAuthorizationManager.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.ContextAttributes;
import se.curity.identityserver.sdk.attribute.SubjectAttributes;
import se.curity.identityserver.sdk.authorization.AuthorizationResult;
import se.curity.identityserver.sdk.authorization.ScimObligation;
import se.curity.identityserver.sdk.authorization.scim.ScimAuthorizationActionAttributes;
import se.curity.identityserver.sdk.authorization.scim.ScimAuthorizationManager;
import se.curity.identityserver.sdk.authorization.scim.ScimAuthorizationResourceAttributes;

public final class ${pluginName}ScimAuthorizationManager implements ScimAuthorizationManager
{
    private final ${pluginName}AuthorizationManagerConfig _configuration;

    public ${pluginName}ScimAuthorizationManager(${pluginName}AuthorizationManagerConfig configuration)
    {
        _configuration = configuration;
    }

    @Override
    public AuthorizationResult<ScimObligation> getScimAuthorizationResult(SubjectAttributes subjectAttributes,
                                                                          ScimAuthorizationActionAttributes scimAuthorizationActionAttributes,
                                                                          ScimAuthorizationResourceAttributes scimAuthorizationResourceAttributes,
                                                                          ContextAttributes contextAttributes)
    {
        // TODO: Perform decision and return appropriate AuthorizationResult
        return AuthorizationResult.notApplicable();
    }

}

File: kotlin-authorization-manager-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__ScimAuthorizationManager.kt
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.ContextAttributes;
import se.curity.identityserver.sdk.attribute.SubjectAttributes;
import se.curity.identityserver.sdk.authorization.AuthorizationResult;
import se.curity.identityserver.sdk.authorization.ScimObligation;
import se.curity.identityserver.sdk.authorization.scim.ScimAuthorizationActionAttributes;
import se.curity.identityserver.sdk.authorization.scim.ScimAuthorizationManager;
import se.curity.identityserver.sdk.authorization.scim.ScimAuthorizationResourceAttributes;

class ${pluginName}ScimAuthorizationManager(private val _configuration:${pluginName}AuthorizationManagerConfig): ScimAuthorizationManager
{
    override fun getScimAuthorizationResult(subjectAttributes: SubjectAttributes ,
                                            scimAuthorizationActionAttributes: ScimAuthorizationActionAttributes,
                                            scimAuthorizationResourceAttributes: ScimAuthorizationResourceAttributes,
                                            contextAttributes: ContextAttributes)
        : AuthorizationResult<ScimObligation>
    {
        // TODO: Perform decision and return appropriate AuthorizationResult
        return AuthorizationResult.notApplicable();
    }

}
